test/parse.js

```diff
@@ -34,31 +34,31 @@ test('parse()', function (t) {
 
     t.test('arrayFormat: brackets allows only explicit arrays', function (st) {
         st.deepEqual(qs.parse('a[]=b&a[]=c', { arrayFormat: 'brackets' }), { a: ['b', 'c'] });
-        st.deepEqual(qs.parse('a[0]=b&a[1]=c', { arrayFormat: 'brackets' }), { a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a[0]=b&a[1]=c', { arrayFormat: 'brackets' }), { 'a[0]': 'b', 'a[1]': 'c' });
         st.deepEqual(qs.parse('a=b,c', { arrayFormat: 'brackets' }), { a: 'b,c' });
-        st.deepEqual(qs.parse('a=b&a=c', { arrayFormat: 'brackets' }), { a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a=b&a=c', { arrayFormat: 'brackets' }), { a: 'c' });
```

Comment:
i see how this seems correct, because it's currently set to brackets but is parsing as repeat.

However, the current behavior matches what PHP, Rails, and express all do, so I'm not sure it should change.

Reply:
Rails and PHP actually parse it as overwriting

this is the reason I've started all this: in my rails app I've switched from "chaos and anarchy " to qs and got one spec failure where the suite was appending params and with qs that lead to an array

Reply:
hmm, thanks for correcting me then. i'll think on this one.